mempool/clist_mempool.go

```diff
@@ -45,6 +45,11 @@ type CListMempool struct {
 
 	// Keeps track of the rechecking process.
 	recheck *recheck
+	// It is true while TXs are being rechecked
+	rechecking atomic.Bool
+	// whether rechecking TXs cannot be completed before
+	// a new block is decided
+	recheckFull atomic.Bool
```

Comment:
I would define this variable inside the `recheck` struct, to keep all the recheck state in one place.